File: backend/Models/User_history.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace backend.Models
{
    public class User_history
    {
        [Key]
        public int Id { get; set; }
        
        public User user { get; set; }
        public Recipe recipe { get; set; }
    }
}

File: backend/main/Models/User_history.java
package main.java.Models;

import javax.persistence.*;

@Entity
public class User_history
{
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int userhistoryId;

    public User_history() {
    }

    public User_history(int userhistoryId) {
        this.userhistoryId=userhistoryId;
    }

    public int getUserhistoryId() {
        return userhistoryId;
    }

    public void setUserhistoryId(int userhistoryId) {
        this.userhistoryId = userhistoryId;
    }

    @OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private User user;

    @OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private Recipe recipe;
}